### DISCUSSION TITLE
Add ability to detect and import extensions through a configuration file

### DISCUSSION THREAD

**Discussion by flevi29** [directus/directus#18138] [Feature Requests]
### Summary

Benefits:
- multiple extensions can be transpiled at once with any directory structure without being forced to use [preserveModules](https://rollupjs.org/configuration-options/#output-preservemodules) in `vite`
- a custom and potentially cleaner directory structure

It is possible even now to transpile multiple extensions at once, and for them to share code, but the resulting file structure is a complete mess with `vite` [preserveModules](https://rollupjs.org/configuration-options/#output-preservemodules), and it's just a hassle in my case to depend on the current very specific directory structure.

### Motivation

My motivation is to make bundling of extensions easier and simpler, by being able to tell Directus where to find them instead of Directus telling me where to put them.

### Detailed Design

- A new environment variable is needed for the location of the config file, by default it is `undefined`, so a new entry is needed for `allowedEnvironmentVars`, perhaps `EXTENSIONS_CONFIG`
- This could be an ESM javascript file with a default export (as the api seems to be heading towards ESM), and might be evaluated by the following `zod` schema:
```typescript
export const ExtensionsConfig = z.record(z.string(), z.discriminatedUnion("type", [
	z.strictObject({
		type: z.enum(NON_HYBRID_EXTENSION_TYPES),
		path: z.string(),
		entrypoint: z.string(),
	}),
	z.strictObject({
		type: z.enum(HYBRID_EXTENSION_TYPES),
		path: z.string(),
		entrypoint: z.strictObject({
			app: z.string(),
			api: z.string(),
		}),
	}),
]));
```
- A new extension getter function is required, perhaps `getLocalExtensionsFromConfig`, this imports the file, validates it's structure and returns its extensions
- Additional watching functionality is required to track this file and the described extensions within
- Maybe the schema type could be exported via `z.infer<>` for use in the configuration file for coding assistance

I have done an implementation of this, if it may ever be approved I can open a pull request.

### Must Have:
- Ability to track and import extensions through a config file

### Should Have:
- N/A

### Could Have:
- N/A

### Won't Have:
- N/A

### Drawbacks

I can't think of any, other than the time it takes to implement it perhaps, but it really doesn't seem that costly.

### Alternatives

Remain with current approach alone.

### Adoption Strategy

It is not a breaking change. Only a small addition. In the future everyone can choose to either use the config or the directory structure, whichever they prefer, but if people are developing locally multiple extensions they might want the config approach.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
